gui-lib/mred/private/wx/gtk/frame.rkt

```diff
@@ -595,7 +599,20 @@
     (unless (num . < . (gdk_screen_get_n_monitors s))
       (fail))
     (gdk_screen_get_monitor_geometry s num r)
-    r))
+    (cond
+      ;; work around the upstream issue https://gitlab.gnome.org/GNOME/gtk/-/issues/2599
+      [(and wayland?
+            (= (gtk_get_major_version) 3)
+            (>= (gtk_get_minor_version) 24)
+            (>= (gtk_get_micro_version) 9))
```

Comment:
I think this `and` probably doesn't do what you intend for a minor version that's strictly more than 24.